lib/cdo/shared_constants.rb

```diff
@@ -0,0 +1,16 @@
+# A set of constants shared between ruby and JS via /shared/config/sharedConstants.json
+require 'json'
+
+module SharedConstants
+  @@json = nil
+
+  def self.json
+    return @@json if @@json
+
+    file = File.read(File.join(__dir__, '../../shared/config/constants.json'))
+    # @@json = JSON.parse(file)
+    @@json = JSON.parse(file, object_class: OpenStruct)
```

Comment:
For color.js, that build step happens as part of the apps build (and would be reasonably easy to imitate here). I wonder exactly where that step happens for the ruby side of things though.